cmd/createorupdate/createorupdate.go

```diff
@@ -130,6 +124,39 @@ func createOrUpdate(ctx context.Context, oc *v20180930preview.OpenShiftManagedCl
 	return oc, nil
 }
 
+func defaultDeployer(ctx context.Context, cs *api.OpenShiftManagedCluster, azuredeploy []byte) error {
```

Comment:
A few points here:

1) I think we should assume the client will use a closure and use that fact to minimise the deployer function signature.

2) I think we should directly pass in the arguments that the client needs to pass through to CreateOrUpdate() (i.e. minimise any bit shuffling the client needs to do).

Therefore I think we should get the deploy function signature down to `func(ctx context.Context, template map[string]interface{}) error`.

I was tempted to include `mode resources.DeploymentMode` as well because it is an essential part of the contract that the client uses `resources.Incremental`.  However doing this is problematic because it ties the signature to a specific Azure API, and the alternative of passing in a string just confuses the issue (IMO).

3) Doing the above should halve the size of defaultDeployer.  At that point, either use a closure to fix things up and call defaultDeployer, or bring all of defaultDeployer inline into createOrUpdate, as you see fit.

```
deployer := func(ctx context.Context, template map[string]interface{}) error {
	return defaultDeployer(ctx, config, template)
}
err = p.CreateOrUpdate(ctx, cs, azuredeploy, oldCs != nil, deployer)
```

Reply:
done